express/src/validators/hiddenNumberMobileOtpValidator.ts

```diff
@@ -0,0 +1,39 @@
+import {NextFunction, Request, Response} from "express";
+import {obscureNumber} from "../lib/mobileNumberUtils";
+import {validateOtp} from "../lib/checkOtp";
+
+type MiddlewareFunction<T, U, V> = (T: Request, U: Response, V: NextFunction) => void;
+
+export function mobileOtpValidator(
+    formActionUrl: string,
+    textMessageNotReceivedUrl: string
+): MiddlewareFunction<Request, Response, NextFunction> {
+    return async (req: Request, res: Response, next: NextFunction) => {
+        let otp: string = req.body["sms-otp"];
+        otp = otp.trim();
+        const mobileNumber = obscureNumber(getMobileNumber(req));
+
+        const validationResponse = validateOtp(otp);
+        if (validationResponse.isValid) {
+            next();
+        } else {
+            res.render("check-mobile.njk", {
+                errorMessages: {smsOtp: validationResponse.errorMessage as string},
+                value: {otp: otp},
+                values: {
+                    mobileNumber: mobileNumber,
+                    formActionUrl: formActionUrl,
+                    textMessageNotReceivedUrl: textMessageNotReceivedUrl
+                }
+            });
+        }
+    };
+}
+
+function getMobileNumber(req: Request): string {
+    if (req.session?.mfaResponse?.codeSentTo) {
+        return req.session.mfaResponse.codeSentTo;
+    } else {
+        return req.session.enteredMobileNumber ? (req.session.enteredMobileNumber as string) : (req.session.mobileNumber as string);
```

Comment:
```suggestion
        return req.session.enteredMobileNumber as string || req.session.mobileNumber as string;
```

This can be simplified